addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py

```diff
@@ -85,10 +85,14 @@ def _get_alternative_values(self):
 
     @api.model
     def _get_alternative_line_value(self, order_line):
+        product_codes = self.env['product.supplierinfo'].search([
+            ('product_tmpl_id', '=', order_line.product_id.product_tmpl_id.id),
+            ('partner_id', '=', self.partner_id.id)
+        ]).mapped('product_code')
```

Comment:
Don't do a search in a method that is called in a for loop, that would be guaranteed slow. Filter on `order_line.product_id.seller_ids` (+ create another variable for the `product_name`). Or even better, do a single search in `_get_alternative_values` that gets the `product.supplierinfo` for all the products that be copied => create a dict of product to bool of whether or not there is a `product_code` or `product_name` for that seller - product combo and pass that to `_get_alternative_line_value` as an arg as a lookup table. This will avoid performance issues for POs with lots of lines in it/

Don't forget to update the `_get_alternative_line_value` method overrides to include the arg if you do pass a lookup table.